modules/src/ics02_client/events.rs

```diff
@@ -0,0 +1,110 @@
+//! Types for the IBC events emitted from Tendermint Websocket by the client module.
+use crate::attribute;
+use crate::events::{IBCEvent, RawObject};
+use crate::ics02_client::client_type::ClientType;
+use crate::ics24_host::identifier::ClientId;
+use anomaly::BoxError;
+
+use serde_derive::{Deserialize, Serialize};
+use std::convert::TryFrom;
+use tendermint::block;
+use tendermint::rpc::event_listener::ResultEvent;
+
+// TODO - find a better place for NewBlock
+#[derive(Debug, Deserialize, Serialize, Clone)]
+pub struct NewBlock {
+    pub height: block::Height,
+}
+
+impl NewBlock {
+    pub fn new(h: block::Height) -> NewBlock {
+        NewBlock { height: h }
+    }
+}
+
+impl TryFrom<&ResultEvent> for NewBlock {
+    type Error = BoxError;
+    fn try_from(result: &ResultEvent) -> Result<Self, Self::Error> {
+        Ok(NewBlock {
+            height: crate::events::extract_block_height(result)?,
+        })
+    }
+}
+
+impl From<NewBlock> for IBCEvent {
+    fn from(v: NewBlock) -> Self {
+        IBCEvent::NewBlock(v)
+    }
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone)]
+pub struct CreateClient {
+    pub height: block::Height,
+    pub client_id: ClientId,
+    pub client_type: ClientType,
+}
+
+impl TryFrom<RawObject> for CreateClient {
+    type Error = BoxError;
+    fn try_from(obj: RawObject) -> Result<Self, Self::Error> {
+        Ok(CreateClient {
+            height: obj.height,
+            client_id: attribute!(obj, "create_client.client_id"),
+            client_type: attribute!(obj, "create_client.client_type"),
+        })
+    }
+}
+
+impl From<CreateClient> for IBCEvent {
+    fn from(v: CreateClient) -> Self {
+        IBCEvent::CreateClient(v)
+    }
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone)]
+pub struct UpdateClient {
+    pub height: block::Height,
```

Comment:
Does it make sense introducing "asbtract" event type with height, client_id and client_type as these three events types have the same fields?

Reply:
Yes, eventually we will. Hopefully for most of the channel and connection also.